InterfaceMATLAB/tsgGetHCoefficients.m

```diff
@@ -34,7 +34,8 @@
         fprintf(1,['Warning: Command had non-empty output:\n']);
         disp(cmdout);
     end
-    [coefficients] = tsgReadMatrix(sFileO);
+    
+    [coefficients] = tsgReadMatrix(sFileO, strcmp(lGrid.sType, 'fourier'));  % read file as a complex file if appropriate
```

Comment:
* Is this the only place where complex numbers are used? If so, you should not change `tsgReadMatrix()` but make the coefficient conversion here.
* MATLAB by default uses `i` for complex number if you have not used it for loop index or something else (if I remember right)
* Also, you are using for loops inside MATLAB which is horribly inefficient, you must always vectorize MATLAB code:
```
if (strcmp(lGrid.sType, 'fourier'))
    coefficients = coefficients(:, 1:2:end) + i * coefficients(:, 2:2:end);
end
```

Reply:
* Were there cases after commit `4a714cd` where I didn't vectorize? I thought I had changed all my old for-loops, but I may have missed one.
* You are right about `i`. I used `sqrt(-1)` just to be extra safe...